test/unit-test/mqtt_agent_command_functions_utest.c

```diff
@@ -73,3 +170,265 @@ int suiteTearDown( int numFailures )
 {
     return numFailures;
 }
+
+/* ========================================================================== */
+
+/**
+ * @brief Test that MQTTAgentCommand_ProcessLoop() works as intended.
+ */
+void test_MQTTAgentCommand_ProcessLoop( void )
+{
+    MQTTAgentContext_t mqttAgentContext;
+    void * pUnusedArg;
+    MQTTAgentCommandFuncReturns_t returnFlags;
+    MQTTStatus_t mqttStatus;
+
+    mqttStatus = MQTTAgentCommand_ProcessLoop( &mqttAgentContext, pUnusedArg, &returnFlags );
+
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
```

Comment:
Validate here that the return flags are set appropriately, and we should do the same for every other function as well.